std/variant.d

```diff
@@ -339,7 +339,17 @@ private:
             static if (target.size < A.sizeof)
             {
                 if (target.type.tsize < A.sizeof)
-                    *cast(A**)&target.store = new A;
+                {
+                    static if (is(A == U[n], U, size_t n))
+                    {
+                        A* p = (new U[n]).ptr;
```

Comment:
The code is fine as shown, distinction being academic. There is no layout difference between one dynamically-allocated `U[n]` and a dynamic allocation of an array of `n` `U` objects.